state_machine/plugs/flight/lib/pycubed.py

```diff
@@ -220,216 +317,339 @@ def RGB(self, value):
+                for f in listdir('/sd/' + folder):
+                    if filen in f:
+                        for i in f.rsplit(filen):
+                            # search .txt files specifically
                             if '.txt' in i and len(i) == 7:
-                                c = i[-7:-4]
+                                c = i[-7: -4]
                                 try:
                                     if int(c) > n:
                                         n = int(c)
                                 except ValueError:
                                     continue
+
                                 if int(i.rstrip('.txt')) > n:
                                     n = int(i.rstrip('.txt'))
                                     break
-                self.filename = '/sd' + _folder + _file + "{:03}".format(n + 1) + ".txt"
-            b = "a"
-            if binary:
-                b = "ab"
-            with open(self.filename, b) as f:
-                f.tell()
+
+                # create new filepath in sd directory, using given
+                # folder/file names
+                self.filename = (
+                    '/sd' + folder + filen + "{:03}".format(n + 1) + ".txt")
+
+            # create new file with open, write timestamp and status
+            with open(self.filename, "a") as f:
+                f.write(
+                    '# Created: {:.0f}\r\n# Status: {}\r\n'.format(
+                        time.monotonic(), self.status))
+
+            # print a confirmation that this new file was created
             print('New self.filename:', self.filename)
             return self.filename
 
     @property
     def storage_stats(self):
-        _sd = 0
+        """
+        return the storage statistics about the SD card and
+        mainboard file system
+        """
+        sd = 0
         if self.hardware['SDcard']:
-            _sd = statvfs('/sd/')
-            _sd = int(100 * _sd[3] / _sd[2])
-        _fs = statvfs('/')
-        _fs = int(100 * _fs[3] / _fs[2])
-        return (_fs, _sd)
+            # statvfs returns info about SD card (mounted file system)
+            sd = statvfs('/sd/')
+            sd = int(100 * sd[3] / sd[2])
+
+        # returns information about the overall file system
+        fs = statvfs('/')
+        fs = int(100 * fs[3] / fs[2])
+
+        # return both sets of information
+        return (fs, sd)
 
     def log(self, msg):
-        if stat(self.logfile)[6] > 1E8:  # 100MB
+        """
+        create/open file and write logs
+        """
+
+        # if size of current open logfile > 100MB, create new log file
+        if stat(self.logfile)[6] > 1E8:
             self.new_Log()
+
+        # open the current logfile and write message msg with a timestamp
         if self.hardware['SDcard']:
             with open(self.logfile, "a+") as file:
                 file.write('{:.1f},{}\r\n'.format(time.monotonic(), msg))
 
     def new_Log(self):
```

Comment:
We may want to consider renaming this, as it breaks the snake_case convention held throughout all other methods.

Reply:
Yeah, I agree, `new_log` would be better

Reply:
Fixed!